node/network/collator-protocol/src/collator_side/advertising.rs

```diff
@@ -0,0 +1,197 @@
+	//	- resolves to groupid - can be found once we know the session (and block height)
+	//	- resolves to `AuthorityDiscoveryIds` - can be found if we have session
+	//	- which resolve to `PeerId`s - can be resolved once connected
+	requested_connections: HashMap<AuthorityDiscoveryId, GroupIndex>,
+	/// `PeerId`s for already established connections for some group.
+	group_peers: HashMap<GroupIndex, HashSet<PeerId>>,
+
+	established_connections: HashMap<Hash, GroupIndex>,
+
+	/// Report metrics.
+	metrics: Metrics,
+}
+
+/// Needed information for establishing connections.
+struct ConnectionInfo {
+	/// The relay parent to use for determining the correct core.
+	relay_parent: Hash,
+	/// Block number to determine the desired backing group.
+	///
+	/// Note: This seemingly redundant info, is not redundant in the case of `pre-connect`. In this
+	/// case the above relay_parent will be used for determining the core in advance, but the
+	/// responsbile backing group will be determined based on the given `block_number`.
+	block_number: BlockNumber,
+	/// What session we are operating in.
+	///
+	/// For `pre-connect`, this will just be a guess, which might be wrong (we assume to stay in
+	/// the same session).
+	session_index: SessionIndex,
+}
+
+/// Information about connections to a validator group.
+///
+/// We keep track of connected peers in a validator group and the messages that should be sent to
+/// them.
+struct GroupConnection {
+	/// Connected peers.
+	peers: HashSet<PeerId>,
+	/// Messages that should be sent to connected peers in this group.
+	///
+	/// When messages are sent, peers might not yet be connected, so we keep all messages that
+	/// should be sent to peers here, if a new peer connects all messages that are still relevant
+	/// to its view are sent.
+	messages: Vec<CollatorProtocolMessage>,
+}
+
+impl Advertiser {
+	pub fn new(local_peer_id: PeerId, collator_pair: CollatorPair, metrics: Metrics) -> Self {
+		Self {
+			local_peer_id,
+			collator_pair,
+			collating_on: None,
+			peer_views: HashMap::new(),
+			view: OurView::default(),
+            required_connections: HashMap::new(),
+			metrics,
+		}
+	}
+
+	/// Ask for additional connections.
+	///
+	/// They will automatically established once a block with block height `when` comes into view
+	/// and will be closed, once it goes out of view, assuming no other entry is present, which
+	/// preserves them.
+	///
+	// - Lookup SessionIndex & Group
+	// - Add to required_connections
+	// - Update Connect message if necessary (connection affects already present block heights)
+	pub fn add_required_connections<Sender: SubsystemSender>(&mut self, sender: &mut Sender, when: BlockNumber, info: ConnectionInfo) {
+		self.requried_connections.insert(when, info);
+	}
+
+	/// Send a message to a validator group.
+	//
+	// - insert message
+	// - send to all already connected peers if in view.
+	//		-> View still relevant with async backing? Likely only when it comes to height.
```

Comment:
View is still relevant - view determines their 'implicit view', which is the view-head + some ancestry (described in more detail here: https://github.com/paritytech/polkadot/issues/5054#issuecomment-1105711569)